velox/common/memory/Memory.h

```diff
@@ -67,87 +67,90 @@ constexpr uint16_t kDefaultAlignment = 64;
 
 class ScopedMemoryPool;
 
-class MemoryPool {
+class MemoryPool : public std::enable_shared_from_this<MemoryPool> {
  public:
-  virtual ~MemoryPool() {}
+  MemoryPool(const std::string& name, std::shared_ptr<MemoryPool> parent);
+  virtual ~MemoryPool();
+
+  ////////////////////////// tree methods ////////////////////////////
+  virtual const std::string& getName() const;
+
+  virtual uint64_t getChildCount() const;
+
+  virtual void visitChildren(
+      std::function<void(MemoryPool* FOLLY_NONNULL)> visitor) const;
+
+  virtual std::shared_ptr<MemoryPool> addChild(
```

Comment:
Would be nice to document all methods and add empty lines between. methods and member variables.

Reply:
Commented.